### PR TITLE
Wip/3mdeb/flashrom

### PR DESCRIPTION
**Pull request by artur-rs**

### CODE REVIEW

**Comment by artur-rs** [comment]
@pietrushnic @miczyg1 @krystian-hebel check if anything should be changed and I will send first patch

File: meson.build [hunk 1]
```diff
@@ -269,6 +269,12 @@ if get_option('plugin_thunderbolt')
   conf.set('HAVE_THUNDERBOLT', '1')
 endif
 
+if get_option('plugin_flashrom')
+  flashrom = find_program('flashrom', required: false)
+  libflashrom = cc.find_library('libflashrom', required: false)
+  conf.set('HAVE_FLASHROM', '1')
```
Comment (miczyg1) [inline meson.build]: I do not quite get it how it works. Both flashrom and libflashrom are not required?
And we set config that we have flashrom?

File: plugins/flashrom/fu-plugin-flashrom.c [hunk 2]
```diff
@@ -51,6 +56,13 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 
 	/* we need flashrom from the host system */
 	data->flashrom_fn = fu_common_find_program_in_path ("flashrom", &error_local);
+	if (flashrom_init(1)) {
```
Comment (miczyg1) [inline plugins/flashrom/fu-plugin-flashrom.c]: What the `1` argument says? Is it needed?

File: plugins/flashrom/fu-plugin-flashrom.c [hunk 3]
```diff
@@ -144,17 +156,30 @@ fu_plugin_update_prepare (FuPlugin *plugin,
...
+		}
+		const size_t flash_size = flashrom_flash_getsize(flashctx);
+		uint8_t *newcontents = malloc(flash_size);
+		if (!newcontents) {
+			g_error ("Error: Out of memory.");
 			return FALSE;
 		}
+
+		// TODO: callback implementation
+		if(flashrom_image_read(flashctx, newcontents, flash_size)) {
```
Comment (miczyg1) [inline plugins/flashrom/fu-plugin-flashrom.c]: Why do we have to read the image in the prepare function?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (artur-rs) [reply meson.build]: my approach is that not everyone wants to use libflashrom/flashrom plugin. `meson` will check if the system has installed above dependencies and set appropriate info during build process

File: plugins/flashrom/fu-plugin-flashrom.c (hunk 2 above)
Comment (artur-rs) [reply plugins/flashrom/fu-plugin-flashrom.c]: maybe I should change to `TRUE` - this argument if is not equal `0`, init performs `selfcheck()`

File: plugins/flashrom/fu-plugin-flashrom.c (hunk 3 above)
Comment (artur-rs) [reply plugins/flashrom/fu-plugin-flashrom.c]: it was originally set like this in the first flashrom-plugin version, I left it "unchanged"

File: plugins/flashrom/fu-plugin-flashrom.c (hunk 2 above)
Comment (miczyg1) [reply plugins/flashrom/fu-plugin-flashrom.c]: Please do, it looks weird now

**Pull request closed without merging**